[WIP] Replace NSIS installer with WiX installer

ItsHarper opened the pull request:
DO NOT MERGE. This still has a ways to go, but now that the changes it is dependent on have been merged into master, I'm opening this pull request as a place for centralized discussion about WiX and the code I've been working on, and to allow GitHub's new code reviews feature to be used.

noahAndrews committed:
Show error state for all errors

noahAndrews committed:
Show error state for all errors

noahAndrews committed:
Merge branch 'master' of https://github.com/jackhumbert/qmk_firmware_flasher into improve_error_detection

noahAndrews committed:
Merge branch 'master' of https://github.com/jackhumbert/qmk_firmware_flasher

noahAndrews committed:
Move success check to correct location

noahAndrews committed:
Check for blank output when resetting

as opposed to checking for one specific error

noahAndrews committed:
Check for correct output when erasing

noahAndrews committed:
Homogenize success checks

noahAndrews committed:
Merge remote-tracking branch 'refs/remotes/origin/improve_error_detection'

noahAndrews committed:
Merge remote-tracking branch 'refs/remotes/jackhumbert/master'

noahAndrews committed:
Create flash_when_ready variable

noahAndrews committed:
Remove code to handle the nonexistant reset button

noahAndrews committed:
Removed code to handle old nonexistant close button

noahAndrews committed:
Cache the JQuery selectors

Better for performance and (IMO) readability

noahAndrews committed:
Update Bootstrap

noahAndrews committed:
Offload  flash button styling to Bootstrap

The button will still be green when disabled this way. It always bugged
me how the button would be red after a successful flash. It made it look
like something was wrong.

noahAndrews committed:
Disable the menubar when running from source

noahAndrews committed:
Fix enableFlashButton method

noahAndrews committed:
Merge branch 'flash-when-ready' into improve-button-behavior

noahAndrews committed:
Added a button to flash when ready

Closes #3

noahAndrews committed:
Get bootstrap from npm

noahAndrews committed:
Fix issue with flash button

During the flash when ready state, the flash button would become enabled
once the keyboard was detected.

noahAndrews committed:
Hide fwr button once the board is detected

noahAndrews committed:
Add electron-debug dependency

Only active when in dev mode. Will allow the use of keyboard shortcuts
to open the dev tools and such.
https://www.npmjs.com/package/electron-debug

noahAndrews committed:
Enable electron-debug

noahAndrews committed:
Fix for Windows packaged version

noahAndrews committed:
Correct dfu_location synchronously

This code wasn't working because it wouldn't change dfu_location until
after dfu_location was accessed.

noahAndrews committed:
Add script to build a Windows package on Windows

noahAndrews committed:
Fixed bad dfu path on Windows packaged version

noahAndrews committed:
Improve FWR button alignment

noahAndrews committed:
Improve window sizing on Windows

tested on Windows 7 and 10

noahAndrews committed:
Improve error on 32-bit Windows

noahAndrews committed:
Add more detail about driver installation to log

noahAndrews committed:
Revert "Improve error on 32-bit Windows"

This reverts commit c66373a69003561c85777afe9401d620990f4b42.

noahAndrews committed:
Run npm install before packaging using batch file

Get all updates

noahAndrews committed:
Merge branch 'improve-button-behavior' into wix-installer

noahAndrews committed:
Enable asar packaging in dist-win32.bat

noahAndrews committed:
WIP of Wix installer

Installs, but there's no wizard, and the driver doesn't install yet

noahAndrews committed:
Switch to 2 wxs files

noahAndrews committed:
Reformat wxs files

noahAndrews committed:
Don't include drivers from other architectures

This will be the main difference between the two wxs files.

noahAndrews committed:
update dist-win32.bat

noahAndrews committed:
Consolidate back to single wxs file

Driver installation should work now! Completely untested on 32-bit
Windows.

noahAndrews committed:
package.js - only log error if there is one

noahAndrews committed:
Add wizard to msi file

noahAndrews committed:
Delete old msi files before building

noahAndrews committed:
Make electron version specific

This is important for WiX installer maintanence, plus it's recommended
for Electron in particular anyway:
http://electron.atom.io/docs/tutorial/electron-versioning/

1.4.12 is out, but it only fixes an issue that doesn't affect us anyway.

noahAndrews committed:
Update license

The rtf file is for the WiX installer.

noahAndrews committed:
Show correct license in WiX installer

noahAndrews committed:
Renove NSIS file

noahAndrews committed:
Clean up repository

noahAndrews committed:
Remove unecessary stuff from asar archive

noahAndrews committed:
Removed faulty RegEx

noahAndrews committed:
Put libusb0.dll in dfu directory

This always copies the 32-bit dll, even on 64-bit systems. Shouldn't
cause any problems.

noahAndrews committed:
Don't list the atmel driver in Programs and Features

noahAndrews committed:
Remove most explicit component GUIDs

noahAndrews committed:
Rename wix folder to msi

We'll have another folder that will contain the files to generate
setup.exe using burn

noahAndrews committed:
Renamed QMK_Firmware_Flasher.wxs

noahAndrews committed:
Use the Burn bootstrapper

noahAndrews committed:
Merge upstream

ItsHarper commented:
On another note, I understand why the old commits are there, but I need to figure out how to stop all of them from showing up in the future, ideally while keeping them intact in my fork.

skullydazed commented:
I can squash before merging. Or you can maintain two branches, one that you work on and one that you periodically merge into and then squash. Your choice as to which way to go.

noahAndrews committed:
Don't uninstall driver

noahAndrews committed:
dist32-bat - Only remove setup executable

noahAndrews committed:
Exclude msi and setup directories from package

noahAndrews committed:
Set installer version dynamically

noahAndrews committed:
Added WixDifxAppExtension.dll

noahAndrews committed:
Delete dist-win32.sh

noahAndrews committed:
Delete dist.sh

noahAndrews committed:
Merge branch 'master' into wix-installer

noahAndrews committed:
Initial  AppVeyor deployment test

noahAndrews committed:
add appveyor.yml

noahAndrews committed:
Appveyor - get version from package.json

noahAndrews committed:
Appveyor fixes

noahAndrews committed:
working on batch script

noahAndrews committed:
Initiate Travis build after AppVeyor finishes

noahAndrews committed:
Package windows version with icon

noahAndrews committed:
Merge branch 'jackhumbert-master' into wix-installer

noahAndrews committed:
Fix Windows icon

noahAndrews committed:
Restore lost packaging modification

noahAndrews committed:
dist-win32.bat is once again ready for testing

noahAndrews committed:
Bring in changes from master branch

noahAndrews committed:
2017 now

noahAndrews committed:
More changes pulled in

noahAndrews committed:
Merge branch 'wix-installer' of https://github.com/NoahAndrews/qmk_firmware_flasher into wix-installer

noahAndrews committed:
0.5.4

noahAndrews committed:
0.5.5

noahAndrews committed:
Make travis build the correct branch

noahAndrews committed:
0.5.6

noahAndrews committed:
move curl to batch file

noahAndrews committed:
0.5.7

noahAndrews committed:
Display appveyor branch in intiate-travis-build

noahAndrews committed:
0.5.8

noahAndrews committed:
Reduce top-level clutter

noahAndrews committed:
add release description to appveyor.yml

noahAndrews committed:
Always attempt to build master branch on Travis

noahAndrews committed:
Notify gitter room about Travis builds

noahAndrews committed:
Include version in setup filename

noahAndrews committed:
set DEBUG variable in travis

noahAndrews committed:
Fix AppVeyor Artifact

noahAndrews committed:
Change AppVeyor packaging script

noahAndrews committed:
Debug electron-builder in Travis

noahAndrews committed:
Remove debug variable

noahAndrews committed:
Attempted fix for Appveyor "error"

noahAndrews committed:
Revert "Attempted fix for Appveyor "error""

This reverts commit b750ac1c3ed08fe074ee15f14d255bcfd19a4e6f.

noahAndrews committed:
Renamed wix folder to windows-installer

noahAndrews committed:
add updater.bat

noahAndrews committed:
Prepare for uploading to Bintray

noahAndrews committed:
0.5.3

noahAndrews committed:
Include bintray-travis in release commit

noahAndrews committed:
Add artifact name in appveyor.yml

noahAndrews committed:
Remove v from version number (Travis)

noahAndrews committed:
WIP Mac update script

noahAndrews committed:
Added electron-rebuild

noahAndrews committed:
CI fixes

noahAndrews committed:
Attempted Travis fix

noahAndrews committed:
package.json fix

noahAndrews committed:
Fix prepare-asar-mac.sh permissions

noahAndrews committed:
Fix prepare-asar-mac.sh

noahAndrews committed:
Fix Bintray uploading from Travis

noahAndrews committed:
Get WiX location dynamically

noahAndrews committed:
Clean up packaging directory

noahAndrews committed:
Use latest version of Node 7 for Appveyor builds

noahAndrews committed:
Use latest version of node
We should be safe doing this, since the version that really matters is the one bundled with Electron.

noahAndrews committed:
fix permissions

noahAndrews committed:
Rename Windows.ico to icon.ico

noahAndrews committed:
Working on package.js

noahAndrews committed:
Block packaging Mac version on Windows

noahAndrews committed:
Only delete the files for the platforms we are building for

noahAndrews committed:
Fix case sensitivity bug

noahAndrews committed:
Fix major packaging bug
